### PR TITLE
business logic and ui for service accounts

### PR DESCRIPTION
**Pull request by oyo**
It's a big PR because i rewrote big parts of the business logic using RTK query. Also I switched local overlays to our Redux controlled overlay system

### CODE REVIEW

File: cx-portal/src/components/pages/Test/index.tsx [hunk 1]
```diff
@@ -1,3 +1,82 @@
...
+} from 'features/admin/serviceApiSlice'
+import { sample } from 'lodash'
+
+const ROLE_IDS = [
+  '607818be-4978-41f4-bf63-fa8d2de51155',
+  '607818be-4978-41f4-bf63-fa8d2de51156',
+  '607818be-4978-41f4-bf63-fa8d2de51157',
+]
+
 export default function Test() {
```
Comment (PI-FerasAhmad) [inline cx-portal/src/components/pages/Test/index.tsx:37]: what is this test file?

File: cx-portal/src/components/overlays/AddTechnicalUser/index.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+  return (
+    <>
+      <DialogHeader
+        title={t('content.addUser.technicalUserHeadline')}
+        intro={t('content.addUser.technicalUserSubheadline')}
+        closeWithIcon={true}
+        onCloseWithIcon={() => dispatch(closeOverlay())}
+      />
+      <DialogContent className="w-100">
```
Comment (PI-FerasAhmad) [inline cx-portal/src/components/overlays/AddTechnicalUser/index.tsx:99]: I think className= w-100 will not working hier, we dont have bootstrap in Portal Project

File: cx-portal/src/components/overlays/AddTechnicalUser/index.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        </Button>
+        <Button
+          variant="contained"
+          onClick={onFormSubmit}
+          disabled={formHasErrors()}
+        >
+          {t('global.actions.confirm')}
+        </Button>
+      </DialogActions>
+      {response && (
```
Comment (PI-FerasAhmad) [inline cx-portal/src/components/overlays/AddTechnicalUser/index.tsx:121]: is these brackets '(' hier needed?

**Review by PI-FerasAhmad — APPROVED** [review]
just some notes, otherwise looks good to me

File: cx-portal/src/components/pages/Test/index.tsx (hunk 1 above)
Comment (oyo) [reply cx-portal/src/components/pages/Test/index.tsx:37]: the test page I introduced to quickly test new features without interfering with existing pages - we will make this accessible for developers only

File: cx-portal/src/components/overlays/AddTechnicalUser/index.tsx (hunk 3 above)
Comment (oyo) [reply cx-portal/src/components/overlays/AddTechnicalUser/index.tsx:121]: these round brackets are not required - but automatically added by the prettier.

**Review by PI-FerasAhmad — APPROVED** [review]

**Pull request merged**